chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,3105 @@
+        all_bytes = read_items(self, serial)
+    elif handshake_result == HandShakeStuts.RadioWrong:
+        raise errors.RadioError("Radio not match")
+    elif handshake_result == HandShakeStuts.PwdWrong:
+        raise errors.RadioError("download redio password Wrong")
+    else:
+        raise errors.RadioError("connect Wrong")
+    exit_programming_mode(self)
+    return memmap.MemoryMapBytes(bytes(all_bytes))
+
+
+def do_upload(self):
+    serial = self.pipe
+    handShake_Result = handshake(self, serial)
+    if handShake_Result == HandShakeStuts.Normal:
+        write_items(self, serial)
+    elif handShake_Result == HandShakeStuts.RadioWrong:
+        raise errors.RadioError("Radio not match")
+    elif handShake_Result == HandShakeStuts.PwdWrong:
+        raise errors.RadioError("download redio password Wrong")
+    else:
+        raise errors.RadioError("connect Wrong")
+    exit_programming_mode(self)
+
+
+def handshake(self, serial):
+    databytes = b""
+    num = 0
+    while num <= 5:
+        serial.write(get_handshake_bytes(self.MODEL + " "))
+        databytes = serial.read(self.page_len)
+        flag, databytes = handle_connect_ver(
+            databytes, serial
+            )
+        if flag:
+            break
+        num += 1
+    return handshake_handle_connect_event(self, databytes)
+
+
+def read_items(self, serial):
+    all_bytes = bytearray(self._memsize)
+    status = chirp_common.Status()
+    status.msg = "Cloning from radio"
+    status.cur = 0
+    status.max = self._memsize
+    for item in Clone_TypeEnum:
+        try:
+            item_bytes = get_read_current_page_bytes(
+                self, item.value, serial, status
+                )
+            if item_bytes:
+                all_bytes = read_item_handle_connect_event(
+                    all_bytes, item_bytes, item
+                    )
+        except Exception as e:
+            LOG.error(
+                f"read item_data error: {item.name} error_msg: {e}"
+                )
+            continue
+    return all_bytes
+
+
+def write_items(self, serial):
+    status = chirp_common.Status()
+    status.msg = "Uploading to radio"
+    status.cur = 0
+    status.max = self._memsize
+    data_bytes = self.get_mmap()
+    for item in Clone_TypeEnum:
+        if (
+            item == Clone_TypeEnum.radioHead
+            or item == Clone_TypeEnum.radioInfo
+            or item == Clone_TypeEnum.radioVer
+            or item == Clone_TypeEnum.zoneData
+            or item == Clone_TypeEnum.outFactoryData
+        ):
+            continue
+        item_bytes = get_write_item_bytes(data_bytes, item)
+        write_item_current_page_bytes(
+            self, serial, item_bytes, item.value, status
+            )
+
+
+def handshake_handle_connect_event(self, dataByte: bytes):
+    if dataByte[14:16] == b"\x00\x01":
+        if dataByte[20] != 1:
+            model_str = (
+                self.MODEL.decode("ascii")
```

Comment:
`MODEL` must be a `str` not a `bytes` so it makes no sense to `decode()` here nor have the `if` condition below.